ivy/functional/ivy/activations.py

```diff
@@ -498,6 +498,32 @@ def softmax(
     return current_backend(x).softmax(x, axis=axis, out=out)
 
 
+def _wrap_between(y, a):
+    """wrap y between [-a, a]"""
+    a = ivy.array(a, dtype=y.dtype)
+    a2 = ivy.array(2 * a, dtype=y.dtype)
+    zero = ivy.array(0, dtype=y.dtype)
+    rem = ivy.remainder(ivy.add(y, a), a2)
+    rem = ivy.where(rem < zero, rem + a2, rem) - a
+    return rem
+
+
+def _softplus_jax_like(
+    x: Union[ivy.Array, ivy.NativeArray],
+    /,
+    *,
+    fn_original=None,
+    beta: Optional[Union[int, float]] = None,
+    threshold: Optional[Union[int, float]] = None,
+    out: Optional[ivy.Array] = None,
+):
+    amax = ivy.relu(x)
+    print(amax)
+    x = ivy.subtract(x, ivy.multiply(amax, ivy.array(2, dtype=x.dtype)))
+    x = ivy.add(amax, ivy.log(ivy.add(1, ivy.exp(x))))
+    x = ivy.real(x) + _wrap_between(ivy.imag(x), ivy.pi).astype(x.dtype) * ivy.astype(1j, x.dtype)
+    return x
```

Comment:
I'm not sure if I understand this implementation here, I thought softplus was simply `ivy.log(ivy.add(1., ivy.exp(x)))`. I also can't see a `beta` or `threshold` in here, and I think we should implement at least `beta` for supersetting, and I think you've left a print statement in by mistake

Reply:
yeah, my bad I forgot to delete the print statement, about softplus(x) = ivy.log(ivy.add(1., ivy.exp(x)) each backend has it's own implementation of softplus also some other frameworks like  doesn't support softplus for complex dtype

Reply:
but torch are going to  make all the activation functions support support complex dtype